azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java

```diff
@@ -114,24 +150,10 @@ public State refresh(State source) {
 
     @Override
     public State clone() {
-      State cloned = new State();
-      cloned.isSynchronized = this.isSynchronized;
-      cloned.startTime = this.startTime;
-      cloned.endTime = this.endTime;
-      cloned.lastUpdatedTime = this.lastUpdatedTime;
-      cloned.numOfTrail = this.numOfTrail;
-      cloned.numOfSuccess = this.numOfSuccess;
-      cloned.numOfFailure = this.numOfFailure;
-      cloned.numOfIgnored = this.numOfIgnored;
-      cloned.isPaused = this.isPaused;
-      cloned.rampStage = this.rampStage;
-      cloned.lastRampDownTime = this.lastRampDownTime;
-      cloned.cachedNumOfTrail = this.cachedNumOfTrail;
-      cloned.cachedNumOfSuccess = this.cachedNumOfSuccess;
-      cloned.cachedNumOfFailure = this.cachedNumOfFailure;
-      cloned.cachedNumOfIgnored = this.cachedNumOfIgnored;
-      cloned.isActive = true;
-      return cloned;
+      return new State(this.rampStage, this.isActive, this.isPaused, this.isSynchronized,
```

Comment:
this works too. What I meant was to use existing constructor to create State object and add the additional values and avoid using empty constructor.